weejson-jackson/src/main/scala/com/rallyhealth/weejson/v1/jackson/FromJson.scala

```diff
@@ -29,32 +29,38 @@ object FromJson extends JsonParserOps {
 
 abstract class JsonParserOps(factory: JsonFactory = DefaultJsonFactory.Instance) {
 
-  def apply(string: String): FromInput = fromParser(factory.createParser(string))
+  def apply(string: String): FromInput = fromReplayableParser(() => factory.createParser(string))
 
-  def apply(bytes: Array[Byte]): FromInput = fromParser(factory.createParser(bytes))
+  def apply(bytes: Array[Byte]): FromInput = fromReplayableParser(() => factory.createParser(bytes))
 
   def apply(in: InputStream): FromInput = fromParser(factory.createParser(in))
 
   def apply(reader: Reader): FromInput = fromParser(factory.createParser(reader))
 
-  def apply(file: File): FromInput = fromParser(factory.createParser(file))
+  def apply(file: File): FromInput = fromReplayableParser(() => factory.createParser(file))
 
-  def apply(path: Path): FromInput = fromParser(factory.createParser(path.toFile))
+  def apply(path: Path): FromInput = fromReplayableParser(() => factory.createParser(path.toFile))
```

Comment:
would a temp file, i.e. an intermediate result from some computation that then gets cleaned up, be an edge case to be concerned about?

in that case, would it simply hit `if (parser.isClosed) throw new Abort("Parser is closed.")`?

Reply:
Good question. Parser probably wouldn't be closed--I think the only way to do that is to call `parser.close()` explicitly. I'd expect a jackson EOF or maybe `IOException` in that case. Regardless, I'm not sure there's anything we should do about it.